### PR TITLE
Feature: Track Info Page

### PR DESCRIPTION
**Pull request by rituraj-wednesday**
Track Info Page created:
- Redirect to Info Page
- Persist data from redux
- If load with link, get Data from API
- Audio Played persist when redirect to info page
- Details of the track

Tack Info Page:
PR6-track-info-page

### CODE REVIEW

File: app/components/Track/index.js [hunk 1]
```diff
@@ -103,6 +105,13 @@ export function Track({ track, currentTrackID }) {
     }
   };
 
+  const openTrackInfoPage = () => {
+    onTrackInfoClick([track]);
+    setTimeout(() => {
+      history.push(`/track/${trackId}`);
+    }, 0);
```
Comment (rajatdangat-ws) [inline app/components/Track/index.js:112]: why do we need a setTimeout?

File: app/containers/TrackInfo/index.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+  const onPlayClick = () => {
+    if (!isTrackPlaying) {
+      audioController.play(trackData, setIsTrackLoading);
+    } else {
+      audioController.stop();
+    }
+  };
+
+  const onDetialIconClick = (url) => {
```
Comment (rajatdangat-ws) [inline app/containers/TrackInfo/index.js]: please fix the typo

File: app/containers/TrackInfo/index.js [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+                >
+                  <PlayCircleFilledRoundedIcon sx={{ color: 'white', fontSize: '3.75rem' }} />
+                </If>
+              </If>
+            </IconButton>
+            <BottomGradient />
+          </TrackContainer>
+          <div>
+            <div>
+              <LabelWrapper>Song Name: </LabelWrapper>
```
Comment (rajatdangat-ws) [inline app/containers/TrackInfo/index.js]: please use the i18n text

File: app/containers/TrackReduxProvider/saga.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+/**
+ * A saga that handles fetching GitHub repositories based on a given repository name.
+ * On success, it dispatches a success action with the fetched data.
+ * On failure, it dispatches a failure action with the error data.
+ *
+ * @date 01/03/2024 - 14:47:28
+ *
+ * @param {Object} action - The action object containing the repository name.
+ * @yields {Effect} The effect of calling the API, and then either the success or failure action.
+ */
```
Comment (rajatdangat-ws) [inline app/containers/TrackReduxProvider/saga.js]: please update the jsdoc comments

File: app/components/Track/index.js (hunk 1 above)
Comment (rituraj-wednesday) [reply app/components/Track/index.js:112]: It is just there for good measure. Letting redux dispatch update then, render with new route.

File: app/containers/TrackInfo/index.js [hunk 5]
```diff
@@ -128,10 +128,10 @@ export function TrackInfo(props) {
   }
 
   useEffect(() => {
-    if (!checkTrackID(trackData.trackId)) {
+    if (!InfoLoading && !checkTrackID(trackData.trackId)) {
```
Comment (rajatdangat-ws) [inline app/containers/TrackInfo/index.js]: why is the variable name Pascal Case? Generally we use Pascal Case for class and components only.

**Review by rajatdangat-ws — APPROVED** [review]

**Pull request merged**